plugins/router/f5/f5.go

```diff
@@ -0,0 +1,1743 @@
+	if err != nil {
+		return false, err
+	}
+
+	_, ok := routes[routename]
+
+	return ok, nil
+}
+
+// Add a rule to the specified policy to route requests to the given hostname to
+// the pool named by poolname.
+func (f5 *f5LTM) addRoute(policyname, routename, poolname, hostname,
+	pathname string) error {
+	success := false
+
+	rulesUrl := fmt.Sprintf("https://%s/mgmt/tm/ltm/policy/%s/rules",
+		f5.host, policyname)
+
+	escapedRulename := f5UrlEscape(routename)
+
+	rulesPayload := struct {
+		// Name of the rule.
+		Name string `json:"name"`
+	}{
+		Name: escapedRulename,
+	}
+
+	err := f5.post(rulesUrl, rulesPayload, nil)
+	if err != nil {
+		if err.(F5Error).httpStatusCode == 409 {
+			glog.V(4).Infof("Warning: Rule %s already exists; continuing with"+
+				" initialization in case the existing rule is only partially"+
+				" initialized...", routename)
+		} else {
+			return err
+		}
+	}
+
+	// If adding the condition or action to the rule fails later on,
+	// delete the rule.
+	defer func() {
+		if success != true {
+			err := f5.deleteRoute(policyname, routename)
+			if err != nil && err.(F5Error).httpStatusCode != 404 {
+				glog.V(4).Infof("Warning: Creating rule %s failed,"+
+					" and then cleanup got an error: %v", routename, err)
+			}
+		}
+	}()
+
+	conditionsUrl := fmt.Sprintf("https://%s/mgmt/tm/ltm/policy/%s/rules/%s/conditions",
+		f5.host, policyname, escapedRulename)
+
+	conditionsPayload := struct {
+		// Name is the name of the condition.  We generally use unique numerals.
+		Name string `json:"name"`
+
+		// CaseInsensitive specifies that string matches are case insensitive.
+		CaseInsensitive bool `json:"caseInsensitive"`
+
+		// HttpHost indicates that the condition must match on vhost.
+		HttpHost bool `json:"httpHost,omitempty"`
+
+		// HttpUri indicates that the condition must match on the request URI.
+		HttpUri bool `json:"httpUri,omitempty"`
+
+		// PathSegment, used with HttpUri, indicates that the condition must match
+		// on a particular path segment from the request URI.
+		PathSegment bool `json:"pathSegment,omitempty"`
+
+		// Index indicates which component of an item (e.g., which segment of
+		// a path) this condition checks.
+		Index int `json:"index"`
+
+		// Equals indicates that the condition tests for equality.
+		Equals bool `json:"equals"`
+
+		// Request indicates that the rule matches on requests as opposed to
+		// responses.
+		Request bool `json:"request"`
+
+		// Host, used with HttpHost, indicates that the rule should match against
+		// the vhost (as opposed to the port or both the vhost and the port).
+		Host bool `json:"host,omitempty"`
+
+		// Values specifies items for matching such as pathnames or hostnames.
+		Values []string `json:"values"`
+	}{
+		Name:            "0",
+		CaseInsensitive: true,
+		HttpHost:        true,
+		Host:            true,
+		Index:           0,
+		Equals:          true,
+		Request:         true,
+		Values:          []string{hostname},
+	}
```

Comment:
Probably better to have a ```types.go``` file with all the F5 request payload structs defined there. Again something for later.